server/test/categoryRouter.test.js

```diff
@@ -0,0 +1,74 @@
+const Category = require('../db/models/category');
+const categoryRouter = require('../api/categoryRouter');
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { getDB } = require('../db');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const mongoose = require('mongoose');
+
+let exampleCategory;
+let defaultUser;
+let mongoServer;
+let User;
+const expect = chai.expect;
+chai.use(chaiHttp);
+
+function handleErr(err) {
+  expect(err).to.be.null;
+}
+
+before(async function () {
+  mongoServer = new MongoMemoryServer();
+  const mongoUri = await mongoServer.getUri();
+  getDB(mongoUri);
+  User = require('../db/models/user');
+  defaultUser = new User({ email: 'lobster-ice-cream-lover@gmeil.com' });
+  defaultUser.save(handleErr);
+  exampleCategory = new Category({ title: 'hello there', userId: defaultUser._id });
+  exampleCategory.save(handleErr);
+});
+
+after(async function () {
+  await mongoose.disconnect();
+  await mongoServer.stop();
+});
+
+it('There should be one user in the database', async function () {
+  await Category.find(function (err, categories) {
+    expect(err).to.be.null;
+    console.log(categories);
+  });
+  const cnt = await User.countDocuments();
```

Comment:
just remembered- no abbreviations for variable names. I'll have to change some of mine too, like 'db'